GNUmakefile

```diff
@@ -1,77 +1,207 @@
+	@echo "==> Building $@..."
+	@CGO_ENABLED=1 GOOS=linux GOARCH=amd64 \
+		go build \
+		-ldflags $(GO_LDFLAGS) \
+		-tags "$(GO_TAGS) lxc" \
+		-o "$@"
+
+# Define package targets for each of the build targets we actually have on this system
+define makePackageTarget
+
+pkg/$(1).zip: pkg/$(1)/nomad
+	@echo "==> Packaging for $(1)..."
+	@zip pkg/$(1).zip pkg/$(1)/*
+
+endef
+
+# Reify the package targets
+$(foreach t,$(ALL_TARGETS),$(eval $(call makePackageTarget,$(t))))
+
+# Only for Travis CI compliance
+.PHONY: bootstrap
+bootstrap: deps
+
+.PHONY: deps
+deps: ## Install build and development dependencies
+	@echo "==> Updating build dependencies..."
+	go get -u github.com/alecthomas/gometalinter
+	gometalinter --install
+	go get -u github.com/kardianos/govendor
+	go get -u golang.org/x/tools/cmd/cover
+	go get -u github.com/axw/gocov/gocov
+	go get -u gopkg.in/matm/v1/gocov-html
+	go get -u github.com/ugorji/go/codec/codecgen
+	go get -u github.com/hashicorp/vault
+	go get -u github.com/a8m/tree/cmd/tree
+
+.PHONY: check
+check: ## Lint the source code
+	@echo "==> Linting source code..."
+	@gometalinter \
+		--deadline 10m \
+		--vendor \
+		--exclude '.*\.generated\.go:\d+:' \
+		--disable-all \
+		--sort severity \
+		$(CHECKS) \
+		./...
+
+generate: LOCAL_PACKAGES = $(shell go list ./... | grep -v '/vendor/')
+generate: ## Update generated code for the current OS/arch
+	@echo "==> Generating code for $(GOOS)..."
+	@go generate $(LOCAL_PACKAGES)
+
+.PHONY: dev
+dev: GOOS=$(shell go env GOOS)
+dev: GOARCH=$(shell go env GOARCH)
+dev: GOPATH=$(shell go env GOPATH)
+dev: DEV_TARGET=pkg/$(GOOS)_$(GOARCH)$(if $(HAS_LXC),-lxc)/nomad
+dev: check ## Build for the current development platform
+	@echo "==> Removing old development build..."
+	@rm -f $(PROJECT_ROOT)/$(DEV_TARGET)
+	@rm -f $(PROJECT_ROOT)/bin/nomad
+	@rm -f $(GOPATH)/bin/nomad
+	@$(MAKE) --no-print-directory \
+		$(DEV_TARGET) \
+		GO_TAGS=nomad_test
+	@mkdir -p $(PROJECT_ROOT)/bin
+	@mkdir -p $(GOPATH)/bin
+	@cp $(PROJECT_ROOT)/$(DEV_TARGET) $(PROJECT_ROOT)/bin/
+	@cp $(PROJECT_ROOT)/$(DEV_TARGET) $(GOPATH)/bin
+
+.PHONY: release
+release: clean check $(foreach t,$(ALL_TARGETS),pkg/$(t).zip) ## Build all release packages which can be built on this platform.
+	@echo "==> Results:"
+	@tree --dirsfirst $(PROJECT_ROOT)/pkg
+
+.PHONY: test
+test: LOCAL_PACKAGES = $(shell go list ./... | grep -v '/vendor/')
+test: dev ## Run Nomad test suites
+	@echo "==> Running Nomad test suites:"
+	@NOMAD_TEST_RKT=1 \
+		go test \
+			-cover \
+			-timeout=900s \
+			-tags="nomad_test $(if $(HAS_LXC),lxc)" \
+			$(LOCAL_PACKAGES)
+
+.PHONY: clean
+clean: GOPATH=$(shell go env GOPATH)
+clean: ## Remove build artifacts
+	@echo "==> Cleaning build artifacts..."
+	@rm -rf "$(PROJECT_ROOT)/bin/"
+	@rm -rf "$(PROJECT_ROOT)/pkg/"
+	@rm -f "$(GOPATH)/bin/nomad"
+
+.PHONY: travis
+travis: ## Run Nomad test suites with output to prevent timeouts under Travis CI
+	@sh -C "$(PROJECT_ROOT)/scripts/travis.sh"
```

Comment:
Can you run generate first on the travis tests. That is why the pretty print tests were failing. I fixed them but I would like the CI to test a realistic build.